src/tools/components/ResultDownload.tsx

```diff
@@ -0,0 +1,166 @@
+  format: ResultFormat[T];
+  description: string;
+  ext: string;
+};
+
+const options: Record<JobTypes, DownloadOptions<JobTypes>[]> = {
+  [JobTypes.ALIGN]: [
+    {
+      format: 'aln-clustal_num',
+      description: 'Resulting alignment',
+      ext: 'txt',
+    },
+    { format: 'out', description: 'Raw output from Align', ext: 'txt' },
+    { format: 'sequence', description: 'Raw submitted sequence', ext: 'fasta' },
+    {
+      format: 'phylotree',
+      description: 'Phylogenetic tree in Newick notation',
+      ext: 'txt',
+    },
+    { format: 'pim', description: 'Percent identity matrix', ext: 'txt' },
+    {
+      format: 'submission',
+      description: 'List of input parameters in XML',
+      ext: 'txt',
+    },
+  ] as DownloadOptions<JobTypes.ALIGN>[],
+  [JobTypes.BLAST]: [
+    { format: 'out', description: 'Raw output from BLAST', ext: 'txt' },
+    { format: 'xml', description: 'XML output', ext: 'xml' },
+    { format: 'json', description: 'JSON output', ext: 'json' },
+    { format: 'ids', description: 'List of matching identifiers', ext: 'txt' },
+    { format: 'accs', description: 'List of matching accessions', ext: 'txt' },
+    {
+      format: 'parameters',
+      description: 'List of input parameters in XML',
+      ext: 'xml',
+    },
+    { format: 'sequence', description: 'Raw submitted sequence', ext: 'fasta' },
+    { format: 'visual-svg', description: 'Visual SVG', ext: 'svg' },
+    {
+      format: 'complete-visual-svg',
+      description: 'Complete visual SVG',
+      ext: 'svg',
+    },
+    { format: 'visual-png', description: 'Visual PNG', ext: 'png' },
+    {
+      format: 'complete-visual-png',
+      description: 'Complete visual PNG',
+      ext: 'png',
+    },
+    { format: 'visual-jpg', description: 'Visual JPG', ext: 'jpg' },
+    {
+      format: 'complete-visual-jpg',
+      description: 'Complete visual JPG',
+      ext: 'jpg',
+    },
+    { format: 'ffdp-query-svg', description: 'FFDP query SVG', ext: 'svg' },
+    { format: 'ffdp-query-png', description: 'FFDP query PNG', ext: 'png' },
+    { format: 'ffdp-query-jpeg', description: 'FFDP query JPG', ext: 'jpg' },
+    { format: 'ffdp-subject-svg', description: 'FFDP subject SVG', ext: 'svg' },
+    { format: 'ffdp-subject-png', description: 'FFDP subject PNG', ext: 'png' },
+    {
+      format: 'ffdp-subject-jpeg',
+      description: 'FFDP subject JPG',
+      ext: 'jpg',
+    },
+  ] as DownloadOptions<JobTypes.BLAST>[],
+  [JobTypes.IDMAP]: [],
+  [JobTypes.PEPTIDE_SEARCH]: [],
+};
+
+type ResultDownloadProps = {
+  jobType: JobTypes;
+  id: string;
+  onToggleDisplay: () => void;
+  nHits?: number;
+  isTableResultsFiltered?: boolean;
+  isTableRowSelected: boolean;
+};
+
+const ResultDownload = memo<ResultDownloadProps>(
+  ({
+    jobType,
+    id,
+    onToggleDisplay,
+    nHits,
+    isTableResultsFiltered,
+    isTableRowSelected,
+  }) => {
+    const [fileFormat, setFileFormat] = useState(options[jobType][0].format);
```

Comment:
This is a little bit of nitpicking really, but I was thinking that if we add a function like `getDefaultOption(jobType)` then we don't have to rely on `options[jobType][0]`. The reason being if we are going to use this in other places, it would be much more reliable to hide the implementation inside a function.

Up to you if you want to keep it the way it is now or not though. It's just a suggestion.